src/devtools/client/debugger/src/components/Editor/Breakpoints/Panel/PanelEditor.js

```diff
@@ -0,0 +1,190 @@
+      return (
+        <div className="edit-actions">
+          {saveButton}
+          {cancelButton}
+        </div>
+      );
+    }
+
+    return (
+      <div className="edit-actions">
+        {cancelButton}
+        {saveButton}
+      </div>
+    );
+  }
+
+  renderForm() {
+    const { logValue, conditionValue } = this.state;
+    const { inputToFocus } = this.props;
+    const hasCondition = conditionValue !== null;
+
+    return (
+      <form>
+        <label htmlFor="logpoint">{`Log message (e.g. "x is", x)`}</label>
+        <div className="input-container">
+          <PanelInput
+            id="logpoint"
+            autofocus={inputToFocus == "logValue" ? true : false}
+            defaultValue={logValue}
+            onChange={cm => this.setState({ logValue: cm.getValue().trim() })}
+            onBlur={this.handleInputBlur}
+            onKeyDown={this.handleKeyDown}
+          />
+          {hasCondition ? (
+            <>
+              <label htmlFor="condition">Condition (e.g. x === true)</label>
+              <PanelInput
+                id="condition"
+                autofocus={inputToFocus == "condition" ? true : false}
+                defaultValue={conditionValue}
+                onChange={cm => this.setState({ conditionValue: cm.getValue().trim() })}
+                onBlur={this.handleInputBlur}
+                onKeyDown={this.handleKeyDown}
+              />
+            </>
+          ) : null}
+          {this.renderActions()}
+        </div>
+      </form>
+    );
+  }
+
+  renderHeader() {
+    const { breakpoint } = this.props;
+    const { conditionValue } = this.state;
+    const hasCondition = conditionValue !== null;
+
+    const addCondition = () => {
+      this.props.setInputToFocus("condition");
+      this.setState({ conditionValue: "" });
+    };
+    const removeCondition = () => {
+      this.props.setInputToFocus("logValue");
+      this.setState({ conditionValue: null });
+    };
+
+    return (
+      <div className="header">
+        <div className="type">Logpoint:</div>
+        <div className="line">Line {breakpoint.location.line}</div>
+        {hasCondition ? (
+          <div className="action" tabIndex="0" onClick={removeCondition}>
+            Remove Condition
+          </div>
+        ) : (
+          <div className="action" tabIndex="0" onClick={addCondition}>
+            Add Condition
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  render() {
+    const { conditionValue, logValue } = this.state;
+    const hasCondition = conditionValue !== null;
+
+    if (!logValue) {
+      // Skip the first render. The state doesn't have the default values for
+      // logValue and conditionValue until after the component mounts.
+      return null;
+    }
+
+    return (
+      <div
+        className={classnames("panel-editor", { conditional: hasCondition })}
+        ref={node => (this.panelEditorNode = node)}
+      >
+        {this.renderHeader()}
```

Comment:
I tried splitting out the Form and Header into their own components, but there was so much plumbing that it wasn't worth it.